### DISCUSSION TITLE
Were there dependency changed? Error happens on `DynamicSidebar.then` on the server.

### DISCUSSION THREAD

**Discussion by donghhan** [fuma-nama/fumadocs#808] [Q&A] [answered]
I tried to make fumadocs UI with `pnpm create fumadocs-app` command and run in the dev mode. But then I found the error

```bash
Internal error: Error: Cannot access DynamicSidebar.then on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
    at Object.get (/Users/apensia/dev/fuma/node_modules/.pnpm/next@14.2.8_react-dom@18.3.1_react@18.3.1/node_modules/next/dist/compiled/next-server/app-page.runtime.dev.js:35:253838)
digest: "645555819"
```

Is there anyone confronting this type of error?

**Comment by fuma-nama** [comment]
What versions of Fumadocs and Next.js are you using? You can open a bug report in the Issues tab and provide more info

**Reply by fuma-nama** [reply]
I've successfully reproduced the problem, Next.js 12.2.8 contains problems on dynamic import with client components. You have to downgrade to 12.2.7 for now

**Reply by donghhan** [reply]
@fuma-nama Thank you for your swift response. Please refer to my `package.json` file below: 

```json
{
  "name": "please-work-shit-man",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "build": "next build",
    "dev": "next dev",
    "start": "next start",
    "postinstall": "fumadocs-mdx"
  },
  "dependencies": {
    "next": "^14.2.7",
    "fumadocs-ui": "13.4.5",
    "fumadocs-core": "13.4.5",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "fumadocs-mdx": "10.0.1"
  },
  "devDependencies": {
    "@types/node": "22.5.2",
    "@types/react": "^18.3.5",
    "@types/react-dom": "^18.3.0",
    "typescript": "^5.5.4",
    "@types/mdx": "^2.0.13",
    "autoprefixer": "^10.4.20",
    "postcss": "^8.4.42",
    "tailwindcss": "^3.4.10"
  }
}
```

Please be advised that I have not modified any versions of NPM pacakges. I just followed [automatic installation](https://fumadocs.vercel.app/docs/ui#automatic-installation) suggested on official documentation. 

I will separately open a bug report for this.

**Comment by donghhan** [comment, marked as answer]
Issue resolved at [here](https://github.com/fuma-nama/fumadocs/issues/809). Closing discussion.
